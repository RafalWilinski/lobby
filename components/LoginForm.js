import { Form, Icon, Input, Button, Checkbox } from "antd";
import Head from "./Head.js";
import HWCenterWrapper from "../components/HVCenterWrapper";

const FormItem = Form.Item;

const LoginForm = props => {
  const { getFieldDecorator } = this.props.form;

  return (
    <Form
      onSubmit={this.props.handleSubmit}
      className="login-form"
      style={{ maxWidth: "280px" }}
    >
      <FormItem>
        {getFieldDecorator("userName", {
          rules: [{ required: true, message: "Please input your username!" }]
        })(
          <Input
            prefix={<Icon type="user" style={{ fontSize: 13 }} />}
            placeholder="Email"
          />
        )}
      </FormItem>
      <FormItem>
        {getFieldDecorator("password", {
          rules: [{ required: true, message: "Please input your Password!" }]
        })(
          <Input
            prefix={<Icon type="lock" style={{ fontSize: 13 }} />}
            type="password"
            placeholder="Hasło"
          />
        )}
      </FormItem>
      <FormItem>
        <a className="login-form-forgot" href="">
          Zapomniałam/em hasła
        </a>
        <Button type="primary" htmlType="submit" className="login-form-button">
          Zaloguj
        </Button>
        Lub <a href="">Zarejestruj sie!</a>
      </FormItem>
    </Form>
  );
};

export default LoginForm;
