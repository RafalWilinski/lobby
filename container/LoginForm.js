import React, { Component } from "react";
import { bindActionCreators } from "redux";
import { Form, Icon, Input, Button, Checkbox } from "antd";

import LoginForm from "../components/LoginForm";

class LoginFormContainer extends Component {
  handleSubmit = e => {
    e.preventDefault();

    this.props.form.validateFields((err, values) => {
      if (!err) {
        console.log("Received values of form: ", values);
      }
    });
  };

  render() {
    return <LoginForm handleSubmit={this.handleSubmit} />;
  }
}

const WrappedLoginForm = Form.create()(LoginFormContainer);

export default WrappedLoginForm;
