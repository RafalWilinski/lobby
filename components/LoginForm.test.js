import React from "react";
import renderer from "react-test-renderer";
import LoginForm from "./LoginForm";

describe("LoginForm", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<LoginForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
