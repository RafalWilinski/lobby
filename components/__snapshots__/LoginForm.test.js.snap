// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders correctly 1`] = `
<form
  className="ant-form ant-form-horizontal login-form"
  onSubmit={[Function]}
  style={
    Object {
      "maxWidth": "280px",
    }
  }
>
  <div
    className="ant-row ant-form-item"
    style={undefined}
  >
    <div
      className="ant-form-item-control-wrapper"
    >
      <div
        className="ant-form-item-control "
      >
        <span
          className="ant-input-affix-wrapper"
          style={undefined}
        >
          <span
            className="ant-input-prefix"
          >
            <i
              className="anticon anticon-user"
              style={
                Object {
                  "fontSize": 13,
                }
              }
            />
          </span>
          <input
            className="ant-input ant-input-lg"
            data-__meta={
              Object {
                "leadingName": "login",
                "name": "login",
                "originalProps": Object {
                  "disabled": false,
                  "id": "login-email",
                  "placeholder": "Email",
                  "prefix": <Icon
                    style={
                      Object {
                        "fontSize": 13,
                      }
                    }
                    type="user"
                  />,
                  "prefixCls": "ant-input",
                  "type": "text",
                },
                "ref": null,
                "rules": Array [
                  Object {
                    "message": "Proszę podac email!",
                    "required": true,
                  },
                ],
                "trigger": "onChange",
                "validate": Array [
                  Object {
                    "rules": Array [
                      Object {
                        "message": "Proszę podac email!",
                        "required": true,
                      },
                    ],
                    "trigger": Array [
                      "onChange",
                    ],
                  },
                ],
                "valuePropName": "value",
              }
            }
            disabled={false}
            id="login"
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Email"
            size="large"
            style={null}
            type="text"
            value=""
          />
        </span>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={undefined}
  >
    <div
      className="ant-form-item-control-wrapper"
    >
      <div
        className="ant-form-item-control "
      >
        <span
          className="ant-input-affix-wrapper"
          style={undefined}
        >
          <span
            className="ant-input-prefix"
          >
            <i
              className="anticon anticon-lock"
              style={
                Object {
                  "fontSize": 13,
                }
              }
            />
          </span>
          <input
            className="ant-input ant-input-lg"
            data-__meta={
              Object {
                "leadingName": "password",
                "name": "password",
                "originalProps": Object {
                  "disabled": false,
                  "id": "login-password",
                  "placeholder": "Hasło",
                  "prefix": <Icon
                    style={
                      Object {
                        "fontSize": 13,
                      }
                    }
                    type="lock"
                  />,
                  "prefixCls": "ant-input",
                  "type": "password",
                },
                "ref": null,
                "rules": Array [
                  Object {
                    "message": "Proszę podac hasło!",
                    "required": true,
                  },
                ],
                "trigger": "onChange",
                "validate": Array [
                  Object {
                    "rules": Array [
                      Object {
                        "message": "Proszę podac hasło!",
                        "required": true,
                      },
                    ],
                    "trigger": Array [
                      "onChange",
                    ],
                  },
                ],
                "valuePropName": "value",
              }
            }
            disabled={false}
            id="password"
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Hasło"
            size="large"
            style={null}
            type="password"
            value=""
          />
        </span>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={undefined}
  >
    <div
      className="ant-form-item-control-wrapper"
    >
      <div
        className="ant-form-item-control "
      >
        <a
          className="login-form-forgot"
          href=""
          style={
            Object {
              "float": "left",
            }
          }
        >
          Zapomniałam/em hasła
        </a>
        <button
          className="ant-btn login-form-button ant-btn-primary ant-btn-lg"
          onClick={[Function]}
          style={
            Object {
              "width": "100%",
            }
          }
          type="submit"
        >
          <span>
            Zaloguj
          </span>
        </button>
        <a
          href="/register"
          onClick={[Function]}
        >
          Zarejestruj sie!
        </a>
      </div>
    </div>
  </div>
</form>
`;
